src/rgw/rgw_string.h

```diff
@@ -125,4 +126,100 @@ static inline StringT create_n_reserve(const size_t reserve_len)
   return ret;
 }
 
+namespace detail {
+
+// variadic sum() to add up string lengths for reserve()
+static inline constexpr size_t sum() { return 0; }
+template <typename... Args>
+constexpr size_t sum(size_t v, Args... args) { return v + sum(args...); }
+
+// traits for string_size()
+template <typename T>
+struct string_traits {
+  static constexpr size_t size(const T& s) { return s.size(); }
+};
+// specializations for char*/const char*
+template <>
+struct string_traits<const char*> {
+  static constexpr size_t size(const char* s) { return std::strlen(s); }
+};
+template <>
+struct string_traits<char*> : string_traits<const char*> {};
+// specializations for null-terminated string literals
+template <std::size_t N>
+struct string_traits<const char[N]> {
+  static constexpr size_t size(const char (&s)[N]) { return N-1; }
```

Comment:
yeah, i think i prefer `strlen()` here too. that's what the `boost::string_view()` constructor does for string literals, so this would make `string_size()` more consistent with those conversions in append/join